### ISSUE TITLE
Inconsistent prop access restriction

### ISSUE THREAD

**Issue by elsassph** [solidjs-community/eslint-plugin-solid#197]
**Describe the bug**

Inside an effect, the restriction to pass a callback reading a prop is inconsistent.

**To Reproduce**

```typescript
createEffect(() => {
    doSomething(() => props.toggle); // <-- solid/reactivity
    
    const toggle = () => props.toggle; // OK
    doSomething(toggle); // OK
```

**Expected behavior**

Either both are OK or both aren't.

**Screenshots**

**Environment (please complete the following information):**

- OS: [e.g. Mac OS 11, Windows 10]
- Node version (`node --version`):
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.5
- `eslint` version (`npm list eslint`/`yarn why eslint`): 8.57.1

**Additional context**

- [ ] I would be willing to contribute a PR to fix this issue

**Comment by ryansolid** [comment]
Confirmed on current main: the inline `doSomething(() => props.toggle)` inside an effect warns while the named-variable form doesn't. The inconsistency is real; the named form slipping through is arguably the false *negative*, but per the precision-first policy the likely resolution is exempting both inside tracked scopes (the effect re-runs regardless). Grouped with the 0.16.1 reactivity precision pass.

**Comment by ryansolid** [comment]
Fixed in v0.16.1. Inline functions passed to unknown calls inside a tracked scope (`doSomething(() => props.toggle)` in an effect) no longer warn, matching the existing behavior for named functions: synchronous calls still run inside the tracked scope, and later calls poll current values like any called function. Known APIs where the rule can prove the callback runs untracked (e.g. `runWithOwner` with a setup-scope owner) still warn.
